### PR TITLE
Update script to recent version of HPP + minor additions

### PR DESCRIPTION
**Pull request by florent-lamiraux**

### CODE REVIEW

File: talos/manipulate_boxes/common_hpp.py [hunk 1]
```diff
@@ -86,18 +86,47 @@ def makeRobotProblemAndViewerFactory(clients):
...
+                      handles = [handles [3], handles [0]], link = True))
+    res.append (Rule (grippers = grippers,
+                      handles = [handles [2], handles [1]], link = True))
+    res.append (Rule (grippers = grippers,
+                      handles = [handles [1], handles [2]], link = True))
+    res.append (Rule (grippers = grippers,
+                      handles = [handles [2], handles [3]], link = True))
+    res.append (Rule (grippers = grippers,
+                      handles = [handles [3], handles [2]], link = True))
+    return res
```
Comment (jmirabel) [inline talos/manipulate_boxes/common_hpp.py:115]: A note for whoever wants to try:

I lately added a procedure to statically validate the graph. It should be able to detect all the issues like *it is not possible to have (left gripper, box/handle1) and (right gripper, box/handle3)*. At the moment, this validation has only been tested on toy problems for testing purposes.
I would be curious to test it on this problem.

Comment (jmirabel) [reply talos/manipulate_boxes/common_hpp.py:115]: The validation procedure should look like:
```python
validation = ps.client.manipulation.problem.createGraphValidation()
problem = ps.client.basic.problem.getProblem()
validation.validate (problem.getConstraintGraph())
if validation.hasErrors() or validation.hasWarnings():
  print (validation.str())
```

Comment (florent-lamiraux) [reply talos/manipulate_boxes/common_hpp.py:115]: It does not seem to validate anything:
```
In [4]: validation.hasErrors ()
Out[4]: False

In [5]: validation.hasWarnings ()
Out[5]: False

In [6]: print (validation.str ())

```

Comment (jmirabel) [reply talos/manipulate_boxes/common_hpp.py:115]: With the rules, it won't. Without the rules, it should. If it doesn't, I will have a look later.

Comment (florent-lamiraux) [reply talos/manipulate_boxes/common_hpp.py:115]: Of course. Without the rules, the procedure finds some collisions. At a first glance, they seem correct.
```
In [6]: print (validation.str ())
Error 0 (talos/left_gripper grasps box/handle1 : talos/right_gripper grasps box/handle3)
  The following collision pairs will always collide.
     Number of collisions : 8.
      Collision between object talos/arm_left_7_link_0 and talos/arm_right_7_link_0
      Collision between object talos/arm_left_7_link_0 and talos/gripper_right_base_link_0
      Collision between object talos/wrist_left_ft_link_0 and talos/wrist_right_ft_link_0
      Collision between object talos/wrist_left_ft_link_0 and talos/wrist_right_ft_tool_link_0
      Collision between object talos/wrist_left_ft_tool_link_0 and talos/wrist_right_ft_link_0
      Collision between object talos/wrist_left_ft_tool_link_0 and talos/wrist_right_ft_tool_link_0
      Collision between object talos/gripper_left_base_link_0 and talos/arm_right_7_link_0
      Collision between object talos/gripper_left_base_link_0 and talos/gripper_right_base_link_0
      
Error 1 (talos/right_gripper > box/handle3 | 0-0_pregrasp)
  The following collision pairs will always collide.
     Number of collisions : 1.
      Collision between object talos/arm_left_7_link_0 and talos/gripper_right_base_link_0
      
Error 2 (talos/left_gripper grasps box/handle2 : talos/right_gripper grasps box/handle4)
  The following collision pairs will always collide.
     Number of collisions : 7.
      Collision between object talos/arm_left_7_link_0 and talos/arm_right_7_link_0
      Collision between object talos/arm_left_7_link_0 and talos/gripper_right_base_link_0
      Collision between object talos/wrist_left_ft_link_0 and talos/wrist_right_ft_link_0
      Collision between object talos/wrist_left_ft_link_0 and talos/wrist_right_ft_tool_link_0
      Collision between object talos/wrist_left_ft_tool_link_0 and talos/wrist_right_ft_tool_link_0
      Collision between object talos/gripper_left_base_link_0 and talos/arm_right_7_link_0
      Collision between object talos/gripper_left_base_link_0 and talos/gripper_right_base_link_0
      
Error 3 (talos/right_gripper > box/handle4 | 0-1_pregrasp)
  The following collision pairs will always collide.
     Number of collisions : 1.
      Collision between object talos/arm_left_7_link_0 and talos/gripper_right_base_link_0
      
Error 4 (talos/left_gripper grasps box/handle3 : talos/right_gripper grasps box/handle1)
  The following collision pairs will always collide.
     Number of collisions : 7.
      Collision between object talos/arm_left_7_link_0 and talos/arm_right_7_link_0
      Collision between object talos/arm_left_7_link_0 and talos/gripper_right_base_link_0
      Collision between object talos/wrist_left_ft_link_0 and talos/wrist_right_ft_link_0
      Collision between object talos/wrist_left_ft_link_0 and talos/wrist_right_ft_tool_link_0
      Collision between object talos/wrist_left_ft_tool_link_0 and talos/wrist_right_ft_tool_link_0
      Collision between object talos/gripper_left_base_link_0 and talos/arm_right_7_link_0
      Collision between object talos/gripper_left_base_link_0 and talos/gripper_right_base_link_0
      
Error 5 (talos/right_gripper > box/handle1 | 0-2_pregrasp)
  The following collision pairs will always collide.
     Number of collisions : 1.
      Collision between object talos/arm_left_7_link_0 and talos/gripper_right_base_link_0
      
Error 6 (talos/left_gripper grasps box/handle4 : talos/right_gripper grasps box/handle2)
  The following collision pairs will always collide.
     Number of collisions : 7.
      Collision between object talos/arm_left_7_link_0 and talos/arm_right_7_link_0
      Collision between object talos/arm_left_7_link_0 and talos/gripper_right_base_link_0
      Collision between object talos/wrist_left_ft_link_0 and talos/wrist_right_ft_link_0
      Collision between object talos/wrist_left_ft_link_0 and talos/wrist_right_ft_tool_link_0
      Collision between object talos/wrist_left_ft_tool_link_0 and talos/wrist_right_ft_tool_link_0
      Collision between object talos/gripper_left_base_link_0 and talos/arm_right_7_link_0
      Collision between object talos/gripper_left_base_link_0 and talos/gripper_right_base_link_0
      
Error 7 (talos/right_gripper > box/handle2 | 0-3_pregrasp)
  The following collision pairs will always collide.
     Number of collisions : 1.
      Collision between object talos/arm_left_7_link_0 and talos/gripper_right_base_link_0
      
Error 8 (talos/left_gripper > box/handle3 | 1-0_pregrasp)
  The following collision pairs will always collide.
     Number of collisions : 1.
      Collision between object talos/gripper_left_base_link_0 and talos/arm_right_7_link_0
      
Error 9 (talos/left_gripper > box/handle4 | 1-1_pregrasp)
  The following collision pairs will always collide.
     Number of collisions : 1.
      Collision between object talos/gripper_left_base_link_0 and talos/arm_right_7_link_0
      
Error 10 (talos/left_gripper > box/handle1 | 1-2_pregrasp)
  The following collision pairs will always collide.
     Number of collisions : 1.
      Collision between object talos/gripper_left_base_link_0 and talos/arm_right_7_link_0
      
Error 11 (talos/left_gripper > box/handle2 | 1-3_pregrasp)
  The following collision pairs will always collide.
     Number of collisions : 1.
      Collision between object talos/gripper_left_base_link_0 and talos/arm_right_7_link_0

```

Comment (jmirabel) [reply talos/manipulate_boxes/common_hpp.py:115]: After oral discussion with @florent-lamiraux , the result is correct.

Thanks for having checked.

**Pull request merged**